src/controllers/messagesController.ts

```diff
@@ -132,4 +135,29 @@ export default class MessagesController {
         .fold<IGetLegalMessageAttachmentResponse>(identity, identity)
         .run()
     );
+
+  public readonly upsertMessageStatus = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorTooManyRequests
+    | IResponseSuccessJson<MessageStatusAttributes>
+  > =>
+    withUserFromRequest(req, async user =>
+      withValidatedOrValidationError(
+        NonEmptyString.decode(req.params.id),
+        messageId =>
+          withValidatedOrValidationError(
+            MessageStatusChange.decode(req.body),
+            change =>
+              this.messagesService.upsertMessageStatus(
```

Comment:
We must remember to use selector in this case in order to rout requests to new messageService

Reply:
Yes, absolutely